tests/ga/test_update.py

```diff
@@ -1670,15 +1673,85 @@ def _mock_popen(cmd, *args, **kwargs):
                     with patch('azurelinuxagent.common.osutil.systemd.is_systemd', return_value=True):
                         update_handler.run(debug=True)
 
-        # Firewall-cmd should only be called 3 times - 1st to check if running, 2nd & 3rd for the QueryPassThrough cmd
-        self.assertEqual(3, len(executed_commands),
-                         "The number of times firewall-cmd should be called is only 3; Executed firewall commands: {0}; All popen calls: {1}".format(
+        # Firewall-cmd should only be called 4 times - 1st to check if running, 2nd, 3rd and 4th for the QueryPassThrough cmd
+        self.assertEqual(4, len(executed_commands),
+                         "The number of times firewall-cmd should be called is only 4; Executed firewall commands: {0}; All popen calls: {1}".format(
                              executed_commands, mock_popen.call_args_list))
         self.assertEqual(PersistFirewallRulesHandler._FIREWALLD_RUNNING_CMD, executed_commands.pop(0),
                          "First command should be to check if firewalld is running")
         self.assertTrue([FirewallCmdDirectCommands.QueryPassThrough in cmd for cmd in executed_commands],
                         "The remaining commands should only be for querying the firewall commands")
 
+    def test_it_should_set_dns_tcp_iptable_if_drop_available_accept_unavailable(self):
+        osutil._enable_firewall = True
+
+        with TestOSUtil._mock_iptables() as mock_iptables:
+            with _get_update_handler(test_data=DATA_FILE) as (update_handler, _):
+                # drop rule is present
+                mock_iptables.set_command(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination), exit_code=0)
+                # non root tcp iptable rule is absent
+                mock_iptables.set_command(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination), exit_code=1)
+                update_handler.run(debug=True)
+
+                drop_check_command = TestOSUtil._command_to_string(osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination))
+                accept_tcp_check_rule = TestOSUtil._command_to_string(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND, mock_iptables.destination))
+                accept_tcp_insert_rule = TestOSUtil._command_to_string(osutil.get_accept_tcp_rule(mock_iptables.wait, AddFirewallRules.INSERT_COMMAND, mock_iptables.destination))
+
+                get_firewall_packets = TestOSUtil._command_to_string(osutil.get_firewall_packets_command(mock_iptables.wait))
```

Comment:
This is an irrelevant command for your use case. You can't keep adding all iptable calls here if we ever add anything in the future. Instead, you should filter out the `mock_iptables.command_calls` array and get only the commands you added and want to verify. Additionally, in this specific use case, the order matters too so make sure of that.

If any doubts of what I'm proposing, please ping me offline and we can get on a quick call.

Reply:
Yes I agree, I now filter the command_calls for only relevant commands. Please let me know your thoughts. In this way the order is maintained.